libff/algebra/curves/curve_utils.tcc

```diff
@@ -31,13 +31,25 @@ GroupT scalar_mul(const GroupT &base, const bigint<m> &scalar)
     return result;
 }
 
+template<typename GroupT>
+decltype(((GroupT *)nullptr)->X) curve_point_y_at_x(
+    const decltype(((GroupT *)nullptr)->X) &x)
+{
+    using base_field = decltype(((GroupT *)nullptr)->X);
+    const base_field x_squared = x * x;
+    const base_field x_cubed = x_squared * x;
+    const base_field y_squared =
+        x_cubed + (GroupT::coeff_a * x) + GroupT::coeff_b;
+    return y_squared.sqrt();
```

Comment:
Don't we have an non terminating function if the square root does not exist?

Reply:
Yes.  `sqrt` will not terminate if `y_squared` is not a square.  The doc comment (in the .hpp) mentions that this does not check for that, and this behaviour matches the existing compressed deserialization code.

Eventually, we could add another `safe` / `unsafe` parameter to the serialization, where `safe` could include this check, as well as the subgroup membership check etc, for a performance cost (this is on my list of issues to add).

For now, it would be convenient to avoid adding extra cost to deserialization, for the purposes of measuring the multi-exp approaches.

Reply:
I remember that we discussed this several times in the past over the phone (i.e. the "safe"/"unsafe" versions for deserialization). I didn't know what "unsafe" meant in this case: It can be unsafe in many ways: return an uninitialized ptr, never return (infinite loop), trigger a segfault etc.

Reply:
"unsafe" only means the function does not check for validity of the data (i.e. the caller is responsible for it).  It's not intended to describe a specific side-effect.  I've removed it from the comment so we can address any ambiguity related safe/unsafe if we add a parameter for it.

Reply:
Got it, thanks